File: frontend/edenBooks/src/modules/orders/services/orderService.ts
import { IOrderService } from "../interfaces/IOrderService";
import { orderApi } from "../api/orderApi";
import { IOrderRequest } from "../interfaces/IOrderRequest";

export const orderService: IOrderService = {
  CreateOrder: async (order: IOrderRequest) =>orderApi.CreateOrder(order),
  GetOrders: async () => orderApi.GetOrders(),
  GetOrderById: async (id: number) => orderApi.GetOrderById(id),
  UpdateOrder: async (id: number, order: IOrderRequest) =>orderApi.UpdateOrder(id, order),
  DeleteOrder: async (id: number) => orderApi.DeleteOrder(id),
};

File: backend/internal/modules/orders/domain/ports/services/order_service.go
package services

import "github.com/javierjpv/edenBooks/internal/modules/orders/domain/entities"


type OrderService interface {
	CheckOrder(order *entities.Order, productIDs []uint) error

	AddOrderIDToProducts(orderID uint, productsIDs []uint) error

	CreateOrder(order *entities.Order, productIDs []uint) error

	UpdateOrder(order *entities.Order) error

	DeleteOrder(id uint) error

	GetOrderByID(id uint) (*entities.Order, error)

	GetFilteredOrders(filters map[string]string) (*[]entities.Order, error)

	ListenPaymentCreated()
}
